src/python/gudhi/representations/vector_methods.py

```diff
@@ -331,8 +343,14 @@ def fit(self, X, y = None):
                 events = np.unique(np.concatenate([pd.flatten() for pd in X] + [[-np.inf]], axis=0))
                 self.grid_ = np.array(events)
             else:
+                self.nan_in_range = np.isnan(np.array(self.sample_range))
+                self.new_resolution = self.resolution
+                if not self.keep_endpoints:
+                    self.new_resolution += self.nan_in_range.sum()
                 self.sample_range = _automatic_sample_range(np.array(self.sample_range), X, y)
-                self.grid_ = np.linspace(self.sample_range[0], self.sample_range[1], self.resolution)
+                self.grid_ = np.linspace(self.sample_range[0], self.sample_range[1], self.new_resolution)
+                if not self.keep_endpoints:
+                    self.grid_ = _trim_endpoints(self.grid_, self.nan_in_range)
```

Comment:
This whole block is common to the three classes t,argeted by this PR and they could be extracted. I think it might improve readability.

This would require moving the first 3 lines from `__init__` to`fit` for `Landscape` and `Silhouette`.

Reply:
Done. I temporarily had a small problem where calling fit twice (possibly forbidden, but happens naturally with `__call__`) would be different from just once, but I fixed that.